### PR TITLE
tdrop: 2018-11-13 -> 2019-10-04

### PR DESCRIPTION
**Pull request by wedens**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by redvers — CHANGES_REQUESTED** [review]
When I try and test this I get the following:

```
[red@apophenia:~/projects/nixpkgs]$ hub pr checkout 72827
Switched to branch 'patch-1'

[red@apophenia:~/projects/nixpkgs]$ nix-shell -I nixpkgs=. -p tdrop
these derivations will be built:
  /nix/store/v9glk5lbdv9dvmj0a3kc4rdb3ql8q16p-source.drv
  /nix/store/bq06y0my14gf2szf3jf67wq5i3sbkm2k-tdrop-unstable-2019-10-04.drv
building '/nix/store/v9glk5lbdv9dvmj0a3kc4rdb3ql8q16p-source.drv'...

trying https://github.com/noctuid/tdrop/archive/60435d240f47719a69ab1f7b2be68ca3ef1df1df.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   155    0   155    0     0    987      0 --:--:-- --:--:-- --:--:--   981
100 19318    0 19318    0     0  54416      0 --:--:-- --:--:-- --:--:-- 54416
unpacking source archive /build/60435d240f47719a69ab1f7b2be68ca3ef1df1df.tar.gz
building '/nix/store/bq06y0my14gf2szf3jf67wq5i3sbkm2k-tdrop-unstable-2019-10-04.drv'...
unpacking sources
unpacking source archive /nix/store/ni0i2c7xnk8fldhg3p17bnd721d8hf40-source
source root is source
patching sources
configuring
no configure script, doing nothing
installing
install flags: SHELL=/nix/store/zavn4np1jvm79f0rafkv0p1mrag09qkz-bash-4.4-p23/bin/bash PREFIX=\$\(out\) install
# 755 is default
install -D -m 755 tdrop "/nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/bin"/tdrop
install -D -m 644 tdrop.1 "/nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/share/man"/man1/tdrop.1
install -D -m 644 LICENSE "/nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/share/licenses"/tdrop/LICENSE
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04
gzipping man pages under /nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/share/man/
strip is /nix/store/cg0k49h66nkdqx6ccwnqr0i4q0fnfznc-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/bin
patching script interpreter paths in /nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04
/nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04/bin/.tdrop-wrapped: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/zavn4np1jvm79f0rafkv0p1mrag09qkz-bash-4.4-p23/bin/bash"
checking for references to /build/ in /nix/store/vajr5lg3vgbd8anaqc95g00jnnj9ga3f-tdrop-unstable-2019-10-04...

[nix-shell:~/projects/nixpkgs]$ tdrop
The program should be in PATH and not contain flags.

```

What am I missing?

**Comment by wedens** [comment]
Interesting... I'll try to reproduce the problem.

**Comment by wedens** [comment]
@redvers Got it. You can't run it without arguments. 

You can run it like this from a terminal where  <command> opens some gui: `tdrop -a auto_hide && <command> && tdrop -a auto_show`. it'll hide a terminal window, open an app with gui and when you close the app, it'll show the same terminal window.

File: pkgs/applications/misc/tdrop/default.nix [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-    wrapProgram $out/bin/tdrop \
-      --prefix PATH : ${lib.makeBinPath [ xwininfo xdotool xprop ]}
+  postInstall = let
+    binPath = lib.makeBinPath [
+      xwininfo
+      xdotool
+      xprop
+      gawk
+      coreutils
+      gnugrep
```
Comment (doronbehar) [inline pkgs/applications/misc/tdrop/default.nix]: Do we definitely need these very standard tools in the `$PATH` as well? It impacts a little bit on the closure size

Comment (wedens) [reply pkgs/applications/misc/tdrop/default.nix]: They're used in the script. Sorry, I don't really understand what you suggest.

Comment (doronbehar) [reply pkgs/applications/misc/tdrop/default.nix]: I mean that these programs are in the `stdenv`'s `$PATH` and in (probably) every user's `$PATH` - hence it shouldn't be necessary to pin them in your derivation just for the sake of making sure such an obvious dependency exists on the system. I suggest removing these 3 lines and hoping that within the lines of regular day-to-day usage, users won't complain.

**Comment by wedens** [comment]
Closed in favour of #94985

**Pull request closed without merging**